platforms/cpu/src/CpuKernels.cpp

```diff
@@ -166,6 +166,12 @@ class CpuCalcForcesAndEnergyKernel::InitForceTask : public ThreadPool::Task {
                 posq[4*i+1] = (float) posData[i][1];
                 posq[4*i+2] = (float) posData[i][2];
             }
+        
+        // Check for invalid positions.
+        
+        for (int i = 4*start; i < 4*end; i++)
+            if (posq[i] != posq[i])
```

Comment:
It won't get optimized away.  This is the standard way to check for a nan.  I'm not sure about inf, but in my testing so far just checking for nan has always been sufficient.